versions/draft-3/SPEC.md

```diff
@@ -1515,6 +1522,130 @@ workflow wf {
 
 In this example, the fully-qualified names that would be exposed as workflow outputs would be `wf.task1.results`, `wf.altname.value`.
 
+## Struct Definition
+A struct is a C-like construct which enables the user to create new compound types that consist of previously existing types. Structs
+can then be used within a `Task` or `Workflow` definition as a declaration in place of any other normal types. The struct takes the place of the
+`Object` type in many circumstances and enables proper typing of its members.
+
+Structs are defined using the `struct` keyword and have a single section consiting of typed declarationns. They are evaluated first prior
+to workflows and tasks.
+```wdl
+struct name { ... }
+```
+
+### Struct Declarations
+The only contents of struct are a set of declarations. Declarations can be any primitive or compound type, as well as other structs, and are defined
+the same way as they are in any other section. The one caveat to this is that declarations within a struct do not allow an expression statement after
+the initial member declaration.
+
+for example the following is a valid struct definition
+```wdl
+struct name {
+    String myString
+    Int myInt
+}
+```
+Whereas the following is invalid
+
+```wdl
+struct invalid {
+    String myString = "Cannot do this"
+    Int myInt
+}
+```
+
+You can also use compound types within a struct to easily encapsulate them within a single object. For example
+```wdl
+struct name {
+    Array[Array[File]] myFiles
+    Map[String,Pair[String,File]] myComplexType
+    String cohortName
+}
+```
+#### Optional and non Empty Struct Values
+Struct declarations can be optional or non-empty (if they are an array type). 
+
+```wdl
+struct name {
+    Arrray[File]+ myFiles
+    Boolean? myBoolean
+}
+```
+
+### Using a Struct
+When using a struct in the declaration section of either a `workflow` or a `task` or `output` section you define them in the same way you would define any other type.
+
+For example, if I have a struct like the following:
```

Comment:
reword to avoid "I"

Reply:
Again, I prefer this to passive-voicing

Reply:
Agree w/ both of you. The "I" sounds super awkward in a spec document.

Mabye "if there is a struct like the following:"?

Reply:
I can’t speak to what one should normally write in a spec document but just keep in mind that passive voice and other lawyerly constructs are a leading source of confusion and difficulty, especially for non-native English speakers.

Reply:
I can’t speak to what one should normally write in a spec document but just keep in mind that passive voice and other lawyerly constructs are a leading source of confusion and difficulty, especially for non-native English speakers.